src/ipc/ipc4/handler.c

```diff
@@ -102,8 +102,7 @@ static int ipc4_comp_params(struct comp_dev *current,
 	return pipeline_for_each_comp(current, ctx, dir);
 }
 
-static int ipc4_pipeline_params(struct pipeline *p, struct comp_dev *host,
-				struct sof_ipc_pcm_params *params)
+static int ipc4_pipeline_params(struct pipeline *p, struct comp_dev *host)
 {
 	struct sof_ipc_pcm_params hw_params;
 	struct pipeline_data data = {
```

Comment:
prefer to remove struct sof_ipc_pcm_params hw_params since params is cleared to zero

Reply:
every component will reassign param from base config, so it doesn't matter if param has random value, 0 or value from upstream component.

Reply:
it is not reasonable use uninitialized parameters